packages/plugins/web3/src/index.js

```diff
@@ -28,9 +28,9 @@ class EmbarkWeb3 {
     this.events.on("blockchain:started", async () => {
       await this.registerWeb3Object();
       this.events.request("embarkjs:console:setProvider", 'blockchain', 'web3', '{web3}');
+      this.events.request("embarkjs:plugin:register", 'blockchain', 'web3', 'embarkjs-web3');
+      await this.events.request2("embarkjs:console:register", 'blockchain', 'web3', 'embarkjs-web3');
```

Comment:
Comment from @iurimatias on Discord:

> this feels incorrect
it should register first, then set provider
also, the register should work, even if no blockchain is startd, it's just registering the plugin
so if ordering if an issue,
then two lines, should be moved above

Reply:
The reason I've moved these two requests inside the `blockchain:started` event is because at least one of them will cause errors when blockchain is disabled and the DApp is built.

`embarkjs:plugin:register` causes Embark to generate EmbarkJS artifacts that expect a generated `blockchain.json` config, which ,in case of a disabled blockchain, won't be around and therefore causes build errors in the pipeline. That's why I decided based on heuristics that, when the chain hasn't started it probably is not even enabled and therefore we don't want to register the plugin so we can prevent the falsy artifact generation.

Reply:
Just to make it a bit more concrete, plugins registered via `embarkjs:plugin:register` a just being stored here: https://github.com/embark-framework/embark/blob/master/packages/stack/embarkjs/src/index.js#L26-L27

And later on, passed to the embarkjs artifact template here: https://github.com/embark-framework/embark/blob/master/packages/stack/embarkjs/src/index.js#L95

Which then generates the code that will break at build time here: https://github.com/embark-framework/embark/blob/master/packages/stack/embarkjs/src/embarkjs-artifact.js.ejs#L34

There's a couple different ways to go about this:

- Instead of relying on the fact whether or not the blockchain has started, we can make use of `blockchainConfig.enabled` and only if it's `true`, we register the plugin.
- We can register it either way, but use `blockchainConfig.enabled` in the artifacts template to decide whether we want the code to be generated or not
- We decide to just always generate the blockchain related (config)code, even when it's disabled

My mind tells me that, when a feature is turned off, we shouldn't generate the related code in the first place.

Thoughts?

Reply:
the order here feels wrong, it needs to register the plugin first `embarkjs:console:register` THEN do `embarkjs:console:setProvider`, I'm suprised ti works otherwise